deepchem/models/tf_new_models/tensor_graph.py

```diff
@@ -0,0 +1,238 @@
+        initial_epoch: epoch at which to start training
+            (useful for resuming a previous training run)
+    checkpoint_interval: int
+      Frequency at which to write checkpoints, measured in epochs
+    """
+    with self.graph.as_default():
+      time1 = time.time()
+      print("Training for %d epochs" % nb_epoch)
+      self.train_op = tf.train.AdamOptimizer(learning_rate).minimize(self.loss)
+      with tf.Session() as sess:
+        sess.run(tf.global_variables_initializer())
+        saver = tf.train.Saver(max_to_keep=max_checkpoints_to_keep)
+        # Save an initial checkpoint.
+        saver.save(sess, self.data_dir, global_step=0)
+        for epoch in range(nb_epoch):
+          avg_loss, n_batches = 0., 0
+          # TODO(rbharath): Don't support example weighting yet.
+          for ind, (X_b, y_b, w_b, ids_b) in enumerate(
+              dataset.iterbatches(batch_size, pad_batches=True)):
+            if ind % log_every_N_batches == 0:
+              print("On batch %d" % ind)
+            feed_dict = {self.features: X_b, self.labels: y_b}
+            fetches = [self.outputs] + [self.train_op, self.loss]
+            fetched_values = sess.run(fetches, feed_dict=feed_dict)
+            loss = fetched_values[-1]
+            avg_loss += loss
+            n_batches += 1
+          if epoch % checkpoint_interval == checkpoint_interval - 1:
+            saver.save(sess, self.data_dir, global_step=epoch)
+          avg_loss = float(avg_loss) / n_batches
+          print('Ending epoch %d: Average loss %g' % (epoch, avg_loss))
+          # Always save a final checkpoint when complete.
+          saver.save(sess, self.data_dir, global_step=epoch + 1)
+      ############################################################## TIMING
+      time2 = time.time()
+      print("TIMING: model fitting took %0.3f s" % (time2 - time1))
+      ############################################################## TIMING
+
+
+class Conv1DLayer(Layer):
+
+  def __init__(self, width, out_channels, **kwargs):
+    self.width = width
+    self.out_channels = out_channels
+    self.out_tensor = None
+    super().__init__(**kwargs)
+
+  def set_parents(self, parents):
+    if len(parents) != 1:
+      raise ValueError("Only One Parent to conv1D over")
+    parent = parents[0]
+    if len(parent.out_tensor.get_shape()) != 3:
+      raise ValueError("Parent tensor must be (batch, width, channel)")
+    parent_shape = parent.out_tensor.get_shape()
+    parent_channel_size = parent_shape[2].value
+    with tf.name_scope(self.name):
+      f = tf.Variable(
+          tf.random_normal([self.width, parent_channel_size, self.out_channels
+                           ]))
+      b = tf.Variable(tf.random_normal([self.out_channels]))
+      t = tf.nn.conv1d(parent.out_tensor, f, stride=1, padding="SAME")
+      t = tf.nn.bias_add(t, b)
+      self.out_tensor = tf.nn.relu(t)
+
+
+class Dense(Layer):
```

Comment:
We should make this `Dense` layer and existing `Dense` layer in `dc.nn.copy` (or `dc.nn.layers`, forget which) agree and then get rid of one.